lib/opsgenie_alert_tagging.rb

```diff
@@ -0,0 +1,104 @@
+require 'httparty'
+require 'dotenv/load'
+require 'pry'
+
+class OpsgenieAlertTagging
+
+  attr_reader :date
+
+  def initialize(date)
+    @date = date
+  end
+
+  def call
+    update_opsgenie
+  end
+
+  def list_all_alerts(date)
+    parse_date = Date.parse(date)
+    format_date = parse_date.strftime("%d-%m-%Y")
+
+    results = HTTParty.get("https://api.opsgenie.com/v2/alerts?query=createdAt%3A#{format_date}&sort=createdAt&order=desc", :headers => {
+      "Content-Type" => "application/json",
+      "Authorization" => "GenieKey #{ENV['OPSGENIE_API_KEY']}"
+    })
+
+    results["data"]
+  end
+
+  def alerts_which_need_updating
+    response = list_all_alerts(date)
+    results = []
+    response.each do |raw_alert|
+      alert = OpsgenieAlert.new(raw_alert)
+      result = {}
+      result[:tags] = alert.tags
+      result[:id] = alert.id
+
+      if bank_holidays.include?(alert.created_at_date) && time_range_wakinghours_weekend.include?(alert.created_at.hour)
+        result[:tags] << "wakinghours"
+        result[:tags] << "OOH"
+
+      elsif time_range_inhours.include?(alert.created_at.hour) && !(alert.created_at.saturday? || alert.created_at.sunday?)
+        result[:tags] << "inhours"
+
+      elsif time_range_wakinghours_weekday.include?(alert.created_at.hour)
+        result[:tags] << "wakinghours"
+        result[:tags] << "OOH"
+
+      elsif time_range_sleepinghours.include?(alert.created_at.hour)
+        result[:tags] << "sleepinghours"
+        result[:tags] << "OOH"
+
+      elsif time_range_wakinghours_weekend.include?(alert.created_at.hour) && (alert.created_at.saturday? || alert.created_at.sunday?)
+        result[:tags] << "wakinghours"
+        result[:tags] << "OOH"
+      end
+
+      result[:tags].uniq!
+      results << result
+    end
+    results
+  end
+
+  def update_opsgenie
+    alerts_which_need_updating.each do |alert|
```

Comment:
im not sure this is actually adding the tags.
```
 $  be ruby opsgenie_alert_tagging.rb 2021-09-19
Updating OpsGenie alert 00777369-66b5-420b-83f3-ea3d1874b203-1632018434930 and the response was ''.
Updating OpsGenie alert 0fda7c1d-02b8-43bc-b16a-548b9255301a-1632018427269 and the response was ''.
Updating OpsGenie alert 4f9af0c8-b1e4-4adf-bb75-50862bbc61b5-1632018254955 and the response was ''.
Updating OpsGenie alert 8964bb95-71eb-4144-b01e-342fe85842f6-1632018247287 and the response was ''.
Updating OpsGenie alert 942eff44-916e-403b-80e3-968dc66531cc-1632018235785 and the response was ''.
```

but when I look at the alert there are no tags.

Reply:
ive worked this out.